### PR TITLE
fix(dashboard): opt-in redirect logic fix

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?

- fixed the dashboard opt-in logic that was running to early
- improved the login page layout

### Screenshots

https://github.com/user-attachments/assets/5984d74d-b25c-46ef-9566-eabe66ea61ee

### CODE REVIEW

File: apps/dashboard/src/components/auth-layout.tsx [hunk 1]
```diff
@@ -2,9 +2,20 @@ import { ReactNode } from 'react';
...
+          <img src={`/images/notifications/notification_01.webp`} alt="logo" className="max-w-[400px]" />
+          <img
+            src={`/images/notifications/notification_02.webp`}
+            alt="logo"
+            className="-mt-[15px] ml-[30px] max-w-[400px]"
+          />
+          <img src={`/images/notifications/notification_03.webp`} alt="logo" className="-mt-[15px] max-w-[400px]" />
+        </div>
+      </div>
+      <div className="flex flex-1 items-center justify-center">{children}</div>
```
Comment (LetItRock) [inline apps/dashboard/src/components/auth-layout.tsx:18]: same layout that we have in the Web app, but improved styles with TW

File: apps/dashboard/src/routes/dashboard.tsx [hunk 2]
```diff
@@ -2,14 +2,17 @@ import { Outlet } from 'react-router-dom';
...
+import { OptInProvider } from '@/context/opt-in-provider';
 
 export const DashboardRoute = () => {
   useMonitoring();
 
   return (
     <ProtectedRoute>
-      <Outlet />
-      <Toaster />
+      <OptInProvider>
```
Comment (LetItRock) [inline apps/dashboard/src/routes/dashboard.tsx:12]: the `OptInProvider` should kick when the user is logged in

File: apps/dashboard/public/images/logo-light.webp
Comment (rifont) [inline apps/dashboard/public/images/logo-light.webp:1]: I didn't see a mention of these logo changes in the PR description, are they intentional?

**Review by rifont — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

File: apps/dashboard/src/components/user-profile.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 import { useNewDashboardOptIn } from '@/hooks/use-new-dashboard-opt-in';
 import { RiSignpostFill } from 'react-icons/ri';
+import { LEGACY_DASHBOARD_URL } from '@/config';
 
 export function UserProfile() {
   const { optOut } = useNewDashboardOptIn();
 
   return (
-    <UserButton afterSignOutUrl={ROUTES.SIGN_IN}>
+    <UserButton afterSignOutUrl={`${LEGACY_DASHBOARD_URL}/auth/login`}>
```
Comment (LetItRock) [inline apps/dashboard/src/components/user-profile.tsx:10]: on logout always redirects to the old web app login page

File: apps/web/src/components/providers/OptInProvider.tsx [hunk 4]
```diff
@@ -20,32 +19,29 @@ export const OptInProvider = ({ children }: { children: React.ReactNode }) => {
...
-      if (NEW_DASHBOARD_ROUTES.some((route) => currentRoute.includes(route))) {
+      if (ROUTES_THAT_REDIRECT_TO_DASHBOARD.some((route) => currentRoute.includes(route))) {
         /**
          * TODO: in order to redirect to the same route, we need to translate the
          * "dev_env_<id>" or wf/step slugs to legacy environment id and vice-versa
          *
          * note: /legacy is part of public URL, so we can't navigate() outside of that
          */
-        window.location.href = window.location.origin;
+        redirectToNewDashboard();
```
Comment (LetItRock) [inline apps/web/src/components/providers/OptInProvider.tsx:41]: when the user is opted-in and is in one of the `/workflows` routes, then redirect to the new dashboard

File: apps/web/src/config/index.ts [hunk 5]
```diff
@@ -94,6 +94,9 @@ export const CLERK_PUBLISHABLE_KEY =
 export const IS_UNDER_DASHBOARD =
   window._env_.REACT_IS_UNDER_DASHBOARD || process.env.REACT_IS_UNDER_DASHBOARD || 'false';
 
+export const WEB_APP_URL =
+  window._env_.REACT_APP_WEB_APP_URL || process.env.REACT_APP_WEB_APP_URL || window.location.origin;
```
Comment (LetItRock) [inline apps/web/src/config/index.ts:98]: when the web application is embedded in the new dashboard we need to know the original web application URL so we can redirect it properly to the login page

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**